documentation/correctness-previous.adoc

```diff
@@ -0,0 +1,130 @@
+The criteria can be visualized as follows.
+
+.The criteria for model correctness as a diagram
+image::../resources/correctness-diagrams/criteria.svg[align="center"]
+
+The discussion of these criteria below is divided into two layers:
+
+. JSON serialization of the model: the _serialization chunk_
+. The model (post-deserialization/in-memory)
+
+Each criterion has three aspects:
+
+. What are the conditions to comply to?
+. How is an incorrectness to be reported?
+. Can an incorrectness be recovered from (and if so: how)?
+
+
+== Serialization chunk
+
+=== Low-level syntactic
+
+The serialization is parsable as JSON.
+Note that “parsable” doesn't necessarily mean “syntax-correct”.
+E.g., trailing commas are not syntax-correct, but don't necessarily prevent the JSON from being parsable, nor deserializable.
+It's desirable to be able to recover from "`simple`"/"`obvious`" syntax errors in the JSON.
+
+Incorrectness is to be reported in terms of the location where parsing the JSON text failed, including an indication whether this was recoverable.
+The reporting is typically dependent on the precise JSON parser used.
+In particular, the JSON parser used might recover automatically, essentially ignoring syntax errors without reporting them in any way.
+This could be problematic, e.g. in case of duplicate keys in an object: the JSON parser is most likely to only reproduce the last value for the duplicate key, ignoring the previous ones.
+
+=== Structural syntactic
+
+The JSON conforms to https://github.com/LionWeb-io/specification/blob/main/serialization/serialization.adoc[the specification for the JSON serialization format].
+Note that JSON Schema isn't powerful enough to express more high-level, nor constraints of a referential nature on the serialization chunk.
+
+Incorrectness is to be reported in terms of e.g. a https://github.com/json-path/JsonPath[JsonPath] to the location(s) of failure(s).
+This is typically dependent on the precise JSON Schema validator framework/library/utility used.
+Recovering from an incorrectness is sometimes possible, e.g.:
+
+. If extra properties are present, they can simply be removed or ignored.
+. If properties are missing, they could just be added, provided they have a compliant default value.
+
+=== Hierarchical
+
+The JSON doesn't violate the following constraints.
+
+. IDs of serialized nodes are unique (within the serialization).
+Violation is to be reported in terms of precise locations in the JSON file of all the serialized nodes with equal ID.
+. The `parent` property of a serialized node matches the `id` of the containing node (whenever that containing node is present in the serialization chunk).
+Concretely: if a node `N` is a child of a node `P` then `N.parent` should equal `P`.
+Violation is to be reported in terms of the ID of the contain**ing** serialized node, the ID of the contain**ed** serialized node, and the `parent` ID stated by that.
```

Comment:
Also for this I think it would be useful if, in addition to specify the ID of the problematic node, the issues would specify the position at which the error has been identified